### PR TITLE
Add GSoC 2023 report

### PR DESCRIPTION
**Pull request by anatoliykmetyuk**

### CODE REVIEW

File: blog/_posts/2023-12-20-gsoc-report.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+layout: blog-detail
+post-type: blog
+by: Anatolii Kmetiuk, Scala Center
+title: "Google Summer of Code 2023 Report"
+---
+We've just wrapped up another fruitful iteration of the Google Summer of Code initiative. This program is a key component of the Scala Center's approach to encouraging more newcomers to contribute to the Scala language, reinforcing its commitment to making Scala contributions more accessible to beginners.
+
+This time around, we had the pleasure of welcoming 6 contributors. Their contributions were directed toward various projects, including data visualization, type-level and asyncronous programming, metaprogramming and more. This article provides a brief overview of their achievements, along with links for further details.
+
+## Unlocking the Full Potential of Scala Web Development
```
Comment (anatoliykmetyuk) [inline blog/_posts/2023-12-20-gsoc-report.md:11]: @akseliristola, @AlexITC, could you please review this section? Feel free to suggest changes, or rewrite it to your liking altogether!

File: blog/_posts/2023-12-20-gsoc-report.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Akseli Ristola worked on integrating Scala 3 support into a [web development template](https://github.com/wiringbits/scala-webapp-template/commits?author=akseliristola) project under the supervision of Alexis Hernandez. This project focused on leveraging the advanced features of Scala 3 to enhance web application development, addressing current limitations and exploring new potentials in Scala-based web frameworks.
+
+Akseli successfully completed two major milestones in this project. The first involved developing a robust integration process for Scala 3, ensuring compatibility and efficiency. The second milestone was the implementation of new features and optimizations specific to Scala 3, which are expected to significantly improve the performance and scalability of web applications developed using Scala.
+
+## http4s Ember WebSocket Client
```
Comment (anatoliykmetyuk) [inline blog/_posts/2023-12-20-gsoc-report.md:16]: @danghieutrung, @armanbilge, could you please review this section? Feel free to suggest changes, or rewrite it to your liking altogether!

File: blog/_posts/2023-12-20-gsoc-report.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+In this project, Dang Trung Hieu worked under the mentorship of Arman Bilge, Christopher Davenport and Diego Alonso to implement a [WebSocket client](https://github.com/http4s/http4s/pull/7196) for the [http4s](https://github.com/http4s/http4s) Ember backend. The goal was to extend the capabilities of the http4s library, a Scala HTTP library, with robust WebSocket support to enable real-time communication features in web applications.
+
+The project's success lies in the development of a fully functional WebSocket client that integrates seamlessly with the existing http4s Ember framework. This integration enhances the library's utility in building modern, real-time web applications, thereby enriching the Scala ecosystem.
+
+## Data Visualization with Doodle
```
Comment (anatoliykmetyuk) [inline blog/_posts/2023-12-20-gsoc-report.md:21]: @danielost, @noelwelsh, could you please review this section? Feel free to suggest changes, or rewrite it to your liking altogether!

File: blog/_posts/2023-12-20-gsoc-report.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+Danylo Ostapchenko worked on this project under the mentorship of Noel Welsh to enhance the [Doodle library](https://www.creativescala.org/doodle/), a tool for compositional vector graphics in Scala. The aim was to improve the library's data visualization capabilities, making it more versatile and user-friendly for developers.
+
+Danylo's work led to significant enhancements in the Doodle library, including new features and optimizations that make data visualization more intuitive and powerful. These improvements are expected to benefit developers seeking to integrate complex graphical representations into their Scala applications. You can read more about Danylo's work in his [final report](https://danielost.github.io/posts/gsoc23/)
+
+## Async-Shifted High-Level APIs
```
Comment (anatoliykmetyuk) [inline blog/_posts/2023-12-20-gsoc-report.md:26]: @alnkravchenko, @rssh, could you please review this section? Feel free to suggest changes, or rewrite it to your liking altogether!

File: blog/_posts/2023-12-20-gsoc-report.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+Olena Kravchenko worked under the mentorship of Ruslan Shevchenko on a project focused on creating an 'async-shifted' version of high-order functions in Scala. The project aimed to streamline asynchronous programming in Scala by making high-level APIs more efficient and easier to use in asynchronous contexts. During this project, Olena pushed the boundaries of the dotty-cps-async library, which provides a framework for asynchronous programming in Scala based on the CPS process algebra. You can read more about her work in the [final report](https://docs.google.com/document/d/1V5harGKumEVvD1975IV-YVEKzyTW1heACfMCg-TdROM/edit).
+
+## Enhancing Scala 3's Macro System
```
Comment (anatoliykmetyuk) [inline blog/_posts/2023-12-20-gsoc-report.md:29]: @zeptometer, @nicolasstucki, could you please review this section? Feel free to suggest changes, or rewrite it to your liking altogether!

File: blog/_posts/2023-12-20-gsoc-report.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+Yuito Murase worked under the mentorship of Nicolas Stucki to enhance the macro system in Scala 3, specifically in the area of pattern matching. The project's goal was to improve quote pattern matching in Scala 3 macros to add better support for matching against types.
+
+Yuito's efforts led to significant improvements in the Scala 3 macro system, particularly in making pattern matching more versatile and robust. These enhancements are crucial for developers who utilize metaprogramming in Scala, allowing for more sophisticated and efficient code generation. You can read more about the work that was done in the [final report](https://gist.github.com/zeptometer/ad414dda221a9f9f99e4135b6b5c1c3c).
+
+## Breaking the Thread Barrier: Boosting I/O Performance in Scala
```
Comment (anatoliykmetyuk) [inline blog/_posts/2023-12-20-gsoc-report.md:34]: @antoniojimeneznieto, @armanbilge, could you please review this section? Feel free to suggest changes, or rewrite it to your liking altogether!

File: blog/_posts/2023-12-20-gsoc-report.md (hunk 1 above)
Comment (AlexITC) [reply blog/_posts/2023-12-20-gsoc-report.md:11]: Looks good.

File: blog/_posts/2023-12-20-gsoc-report.md (hunk 3 above)
Comment (danielost) [reply blog/_posts/2023-12-20-gsoc-report.md:21]: Looks good to me!

File: blog/_posts/2023-12-20-gsoc-report.md (hunk 4 above)
Comment (rssh) [reply blog/_posts/2023-12-20-gsoc-report.md:26]: Olena Kravchenko worked under the mentorship of Ruslan Shevchenko on a project focused on automatically generation of 'async-shifted' version of high-order functions in Scala. The project aimed to streamline asynchronous programming in Scala by making high-level APIs more efficient and easier to use in asynchronous contexts. During this project, Olena pushed the boundaries of the dotty-cps-async library, which provides a framework for asynchronous programming in Scala based on the CPS process algebra. You can read more about her work in the [project overview](https://github.com/rssh/dotty-cps-async/discussions/79) and  [final report](https://docs.google.com/document/d/1V5harGKumEVvD1975IV-YVEKzyTW1heACfMCg-TdROM/). .

File: blog/_posts/2023-12-20-gsoc-report.md (hunk 3 above)
Comment (noelwelsh) [reply blog/_posts/2023-12-20-gsoc-report.md:21]: Me too!

File: blog/_posts/2023-12-20-gsoc-report.md (hunk 5 above)
Comment (nicolasstucki) [reply blog/_posts/2023-12-20-gsoc-report.md:29]: LGTM

Comment (zeptometer) [reply blog/_posts/2023-12-20-gsoc-report.md:29]: Me too!

File: blog/_posts/2023-12-20-gsoc-report.md (hunk 4 above)
Comment (alnkravchenko) [reply blog/_posts/2023-12-20-gsoc-report.md:26]: Seems fine to me

File: blog/_posts/2023-12-20-gsoc-report.md (hunk 6 above)
Comment (antoniojimeneznieto) [reply blog/_posts/2023-12-20-gsoc-report.md:34]: Daniel Spiewak is missing from the list of mentors. Besides that, it looks good to me. 🙂

**Pull request merged**